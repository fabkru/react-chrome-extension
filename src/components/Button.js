import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  constructor(props) {
    super(props);
  }

  getClassNames() {
    let defaultClasses = 'ui button';
    if (this.props.additionalClassNames) {
      return defaultClasses + ' ' + this.props.additionalClassNames;
    }
    return defaultClasses;
  }

  render() {
    if (this.props.isVisible === false) {
      return null;
    }

    return (
      <button className={this.getClassNames()} onClick={this.props.onClick}>
        {this.props.children}
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  isVisible: PropTypes.bool,
  additionalClassNames: PropTypes.string
};

export default Button;
