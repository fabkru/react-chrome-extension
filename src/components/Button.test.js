import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
	const tree = renderer
		.create(
			<Button
				content={'Test'}
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('does not render when isVisible is false', () => {
  const tree = renderer
    .create(
      <Button
        content={'Test'}
        isVisible={false}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
