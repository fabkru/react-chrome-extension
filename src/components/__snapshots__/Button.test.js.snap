// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="ui button undefined"
  onClick={undefined}
>
  Test
</button>
`;

exports[`does not render when isVisible is false 1`] = `null`;